### PR TITLE
feat: add Theme Config Editor

### LINKED ISSUE #39620: Ant Design Theme Editor theme upload
### What problem does this feature solve?

I was hoping to request the addition of a feature to the theme editor (https://ant.design/theme-editor), where by a user can upload their theme config to make editing easier

### What does the proposed API look like?

Soem sort of upload/import button that allows for a json object to be added

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix #39620 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     add Theme Editor theme upload      |
| 🇨🇳 Chinese |    添加主题编辑器主题上传功能       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by rohanbhangui** [comment]
can we make this a window that pops up that allows you to add plain text to a text area. Similar to how you copy the text area when exporting

**Comment by MadCcc** [comment]
> can we make this a window that pops up that allows you to add plain text to a text area. Similar to how you copy the text area when exporting

Really good idea.

**Comment by BoyYangzai** [comment]
@rohanbhangui @MadCcc 
I made the second version, you can experience it
0c80512c829d262557ea3a6574ed660c
a46cec965356c23b42d2fbbb0a52c697
82add1bac89ca7eae2ab5b89094b0ef8

**Comment by rohanbhangui** [comment]
@BoyYangzai  works well! Can we also get the theme editor indexed? It is very difficult to google for. or find. I stumbled on it by accident.

**Comment by BoyYangzai** [comment]
> @BoyYangzai works well! Can we also get the theme editor indexed? It is very difficult to google for. or find. I stumbled on it by accident.

glad for you approve 😊
I also think It's actually kind of hard to find
but I am not a member of `Antd`
@li-jia-nan @MadCcc 
what do you think？

File: .dumi/pages/theme-editor/index.tsx [hunk 1]
```diff
@@ -103,6 +116,47 @@ const CustomTheme = () => {
     setTheme({});
   };
 
+  const [editModelOpen, setEditModelOpen] = useState<boolean>(false);
+  const [editThemeFormatRight, setEditThemeFormatRight] = useState<boolean>(true);
+  const [content, setContent] = useState<TextContent>({
+    text: '{}',
```
Comment (li-jia-nan) [inline .dumi/pages/theme-editor/index.tsx:122]: 初始化的时候能不能让默认值是当前的配置，而不是空对象？

File: package.json [hunk 2]
```diff
@@ -154,7 +154,8 @@
     "rc-util": "^5.25.2",
     "scroll-into-view-if-needed": "^3.0.3",
     "shallowequal": "^1.1.0",
-    "throttle-debounce": "^5.0.0"
+    "throttle-debounce": "^5.0.0",
+    "vanilla-jsoneditor": "^0.11.4"
```
Comment (li-jia-nan) [inline package.json:158]: 这个是不是可以放进 devDependencies 里面？

File: .dumi/pages/theme-editor/index.tsx [hunk 3]
```diff
@@ -1,19 +1,27 @@
-import React, { useEffect } from 'react';
+import { isObject } from 'lodash';
```
Comment (li-jia-nan) [inline .dumi/pages/theme-editor/index.tsx:1]: 这个干掉吧，lodash 太大了，可以自己实现

File: .dumi/pages/theme-editor/index.tsx (hunk 1 above)
Comment (BoyYangzai) [reply .dumi/pages/theme-editor/index.tsx:122]: > 初始化的时候能不能让默认值是当前的配置，而不是空对象？

可以

File: .dumi/pages/theme-editor/index.tsx (hunk 3 above)
Comment (BoyYangzai) [reply .dumi/pages/theme-editor/index.tsx:1]: > 这个干掉吧，lodash 太大了，可以自己实现

ok

File: package.json (hunk 2 above)
Comment (BoyYangzai) [reply package.json:158]: > 这个是不是可以放进 devDependencies 里面？

应该不可以吧qwq

**Comment by BoyYangzai** [comment]
@li-jia-nan 
1.同步了 `Config`，打开即当前 `Config`
2.替代 `Lodash`

File: .dumi/pages/theme-editor/components/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable import/prefer-default-export */
+export function isObject(target: any) {
+  return Object.prototype.toString.call(target) === '[object Object]';
```
Comment (li-jia-nan) [inline .dumi/pages/theme-editor/components/utils.tsx:3]: image

这个好像没办法作为 lodash 的平替吧，lodash 还判断了数组什么的

Comment (BoyYangzai) [reply .dumi/pages/theme-editor/components/utils.tsx:3]: 这里我用错了，只想要纯Object，所以这个就够了，不需要_.isObject

Comment (li-jia-nan) [reply .dumi/pages/theme-editor/components/utils.tsx:3]: OK，那就更新不需要 lodash 了，因为 antd 中用到 lodash 的地方不多，所以后面计划把这个库干掉，体积实在太大了

Comment (BoyYangzai) [reply .dumi/pages/theme-editor/components/utils.tsx:3]: > OK，那就更不需要 lodash 了，因为 antd 中用到 lodash 的地方不多，所以后面计划把这个库干掉，体积实在太大了

哈哈哈，看到你们的pr了

File: package.json (hunk 2 above)
Comment (afc163) [reply package.json:158]: 应该放 devDependencies 里

**Comment by MadCcc** [comment]
导出功能应该不需要了，可以合并起来叫 代码/Code

File: .dumi/pages/theme-editor/components/JSONEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { JSONEditor as Editor, Mode, type JSONEditorPropsOptional } from 'vanilla-jsoneditor';
```
[the remark this answers is not in the document]
Comment (BoyYangzai) [reply .dumi/pages/theme-editor/components/JSONEditor.tsx:1]: > vanilla-jsoneditor 可以加到 https://ant.design/docs/react/recommendation-cn 里来

收到🫡

**Comment by BoyYangzai** [comment]
@MadCcc 
image
这几个文字描述呢，有什么要求吗

**Comment by BoyYangzai** [comment]
@li-jia-nan @afc163 
1.编辑器依赖加入dev
2.移除导出功能，整体更改为代码/code
3.加入https://ant.design/docs/react/recommendation-cn 工具包link

**Comment by MadCcc** [comment]
ping

**Review by MadCcc — APPROVED** [review]

**Pull request merged**